### DISCUSSION TITLE
Execute function by clicking

### DISCUSSION THREAD

**Discussion by Abstrakts** [Budibase/budibase#4920] [Help]
Hello, I am looking for how to execute a javascript function when clicking on an icon.

I would like to compare 2 data models to be able to modify the appearance of the button on click.. would you have an idea please?

**Comment by mjashanks** [comment]
Hi There.

I'd suggest using the `Update State` action upon click. So...

1. If your "Define Actions" for the icon, add `Update State`
2. Set your key to whatever makes sense - let say `toggle`
3. I've set the value, using javascript, to `return true;` - but you can return whatever value makes sense in your scenario
4. Now, save the action.
5. So, everwhere else on the screen, you should have the binding `{{ State.toggle }}` or `$("State.toggle")` in JS available to use for conditions (i.e. set colours when toggle = true)

**Reply by hendrikjap** [reply]
@mjashanks , what if I want to execute print function? 

Can I refer that button id in the JS?

For example, 
const printBtn = document.querySelector(’#print-button’);
$(printBtn).click(function() {
window.print();
});

**Reply by hendrikjap** [reply]
> @mjashanks , what if I want to execute print function?
> 
> Can I refer that button id in the JS?
> 
> For example, const printBtn = document.querySelector(’#print-button’); $(printBtn).click(function() { window.print(); });

**Reply by mjashanks** [reply]
hey @hendrikjap you do not have access to the DOM from Javascript in Budibase. 

It's used for manipulating data for bindings only.

So - this wont be possible.

**Reply by hendrikjap** [reply]
Thanks!
